examples/planning/generate_keyword_ideas.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This example generates keyword ideas from a list of seed keywords."""
+
+from __future__ import absolute_import
+
+import argparse
+import six
+import sys
+from google.ads.google_ads.client import GoogleAdsClient
+from google.ads.google_ads.errors import GoogleAdsException
+
+# Location IDs are listed here: https://developers.google.com/adwords/api/docs/appendix/geotargeting
+# and they can also be retrieved using the GeoTargetConstantService as shown
+# here: https://developers.google.com/google-ads/api/docs/targeting/location-targeting
+_DEFAULT_LOCATION_IDS = '1023191'  # location ID for New York, NY
+# A language criterion ID. For example, specify 1000 for English. For more
+# information on determining this value, see the below link:
+# https://developers.google.com/adwords/api/docs/appendix/codes-formats#languages.
+_DEFAULT_LANGUAGE_ID = '1000'  # language ID for English
+
+
+def main(client, customer_id, location_ids, language_id, keywords, page_url):
+    keyword_plan_idea_service = client.get_service('KeywordPlanIdeaService',
+                                                   version='v1')
+    keyword_competition_level_enum = (
+        client.get_type('KeywordPlanCompetitionLevelEnum', version='v1')
+        .KeywordPlanCompetitionLevel)
+    keyword_plan_network = client.get_type(
+        'KeywordPlanNetworkEnum', version='v1').GOOGLE_SEARCH_AND_PARTNERS
+    locations = map_locations_to_string_values(client, location_ids)
+    language = map_language_to_string_value(client, language_id)
+
+    # Only one of these values will be passed to the KeywordPlanIdeaService
+    # depending on whether keywords, a page_url or both were given.
+    url_seed = None
+    keyword_seed = None
+    keyword_url_seed = None
+
+    # Either keywords or a page_url are required to generate keyword ideas
+    # so this raises an error if neither are provided.
+    if not (len(keywords) or page_url):
```

Comment:
a slightly better form is to remove the redundant len(), ie. if not (keywords or page_url):

see http://google.github.io/styleguide/pyguide.html#2144-decision

Reply:
Oh nice I didn't know an empty array is Falsy in this way.